.github/workflows/PullRequestBuild.yml

```diff
@@ -0,0 +1,28 @@
+name: "Pull Request Build"
+
+env:
+  SOLUTION: 'src/CSESoftware.Repository.SqlBulkRepositoryCore.sln'
+
+on:
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+    name: Build and Test
+    runs-on: windows-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      
+    - name: Nuget Restore for App
+      run: dotnet restore "${{ env.SOLUTION }}"
+  
+    - name: Build Solution
+      run: dotnet publish "${{ env.SOLUTION }}" --configuration Release
+      env:
+        CI: false
+
+    - name: Run Unit Tests on Solution
+      run: dotnet test "${{ env.SOLUTION }}" --no-build --configuration Release --verbosity normal
```

Comment:
The unit tests require an actual database rather than a mock one, since they validate performance of queries on the amount of data and their relationships. How is that going to work in an action or pipeline? Won't the tests fail when they can't establish a connection to the database and prevent merging?